### PR TITLE
Building a Multi-class Text Classification model using H2O and Sckit-learn

### LINKED ISSUE #6725: [Machine Learning] Building a Multi-class Text Classification model using H2O and Sckit-learn
#### Proposed title of article
[Machine Learning] Building a Text Classification model using H2O and Sckit-learn

#### Proposed article introduction
Text classification is a machine learning technique that assigns a set of predefined categories to open-ended text. Based on the incoming text we create a model to learn on the target label and finally predict on the target label. Text classification is one of the fundamental tasks in natural language processing with applications such as sentiment analysis, topic labeling, spam detection, and intent detection.

Multiclass Classification is a classification task with more than two classes. For example classify a set of images of fruits which may be oranges, apples, or pears. Multi-class classification makes the assumption that each sample is assigned to one and only one label: a fruit can be either an apple or a pear but not both at the same time.

In this tutorial, we will build a model, that classifies Consumer Finance Complaints into 5 pre-defined classes. The Consumer Complaint Database is a collection of complaints about consumer financial products and services. We will use Scikit-learn for text preprocessing and vectorization. We will use H2O to automate the NLP process using a pipeline. H2O library will select the best algorithm and perform the model evaluation. 

#### Key takeaways 
1. What is multi-class text classification?
2. Applications of text classification such as sentiment analysis,  topic labeling, spam detection, and intent detection.
3. Text pre-processing (stemming and removing stop words)
3. Text vectorization using TfidfVectorizer.
4.  Using H20 AutoML to select the best algorithm.
5. Using the selected best algorithm to train a customer complaints model.

#### Article quality
This tutorial is unique because will integrate Scikit-learn and H2O in building the text classification model. We will use Scikit-learn for text preprocessing, text cleaning, and vectorization. H20 will create a pipeline that automates the process of model building and select the best algorithm to train a customer complaints model. We will create an end-to-end NLP pipeline starting from cleaning text data, model selection, model evaluation, handling imbalanced datasets.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by charles721**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6725

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

File: content/articles/building-a-multi-class-text-classification-model-using-h2o-and-sckit-learn/index.md [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+```python
+df['Company'].value_counts()
+```
+The company output:
+
+![Company output](/engineering-education/building-a-multi-class-text-classification-model-using-h2o-and-sckit-learn/company-output.png)
+
+The image shows the companies with customer complaints.
+
+#### Renaming `Consumer complaint narrative` column
```
Comment (mercymeave) [inline content/articles/building-a-multi-class-text-classification-model-using-h2o-and-sckit-learn/index.md:154]: Why you please help the readers understand why you perform renaming?

**Review by mercymeave — CHANGES_REQUESTED** [review]
Hello @charles721 

I had problems running your code on Colab, I kept getting the error, "session crashed after using all available ram"

 Would you please do something about this?

**Comment by charles721** [comment]
Hello @mercymeave, about this error "session crashed after using all available ram", I was running ten models that is why it was taking much  RAM. It also takes a lot of time if the process manages to run to completion (about three hours).

So I have made some adjustments and now I am running five models. I have added some new explanations in the article files to meet these new changes. 

Also, I have added a section that explains to the reader how to connect to Google's Colab GPU. The GPU is faster and it will save time running the five models. Without the GPU it took about 1.5hrs.

I have passed some new parameters. I have explained to the reader the reason for doing so. We were using a very large dataset, the new parameters will ensure we only use a portion of the dataset so that the whole process runs to completion without the session crashing.

I have rerun the whole model it took about 46 minutes to run the five models.  You can try to run the code in Google Colab now that I have made the adjustments.

Thanks.

runned-five-models

**Pull request merged**